HorizonEngine/Platform/OpenGL/GLWindow.cpp

```diff
@@ -52,77 +52,78 @@ namespace Hzn
 		//! sets callback to the window close event
 		glfwSetWindowCloseCallback(m_Window, [](GLFWwindow* window)
 			{
-				WindowData data = *(WindowData*)glfwGetWindowUserPointer(window);
+				WindowData* data = (WindowData*)glfwGetWindowUserPointer(window);
 				WindowCloseEvent eventObj;
-				data.callback(eventObj);
+				data->callback(eventObj);
```

Comment:
Why did you change the data callbacks from accessing via an instance( . ) to accessing them via a pointer (->)